packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -1,26 +1,87 @@
-import { ISankeyChartProps, ISankeyChartStyleProps, ISankeyChartStyles } from './SankeyChart.types';
+import { ISankeyChartData, ISankeyChartProps, ISankeyChartStyleProps, ISankeyChartStyles } from './SankeyChart.types';
 import { IProcessedStyleSet } from '@fluentui/react/lib/Styling';
 import * as d3Sankey from 'd3-sankey';
+import { area as d3Area, curveBumpX as d3CurveBasis } from 'd3-shape';
+import { sum as d3Sum } from 'd3-array';
+import { ChartHoverCard, IBasestate, SLink, SNode } from '../../index';
+import { Callout, DirectionalHint } from '@fluentui/react/lib/Callout';
+import { select, selectAll } from 'd3-selection';
+import { FocusZone, FocusZoneDirection, FocusZoneTabbableElements } from '@fluentui/react-focus';
+import { IMargins } from '../../utilities/utilities';
+
 const getClassNames = classNamesFunction<ISankeyChartStyleProps, ISankeyChartStyles>();
+const PADDING_PERCENTAGE = 0.3;
 
-export class SankeyChartBase extends React.Component<
-  ISankeyChartProps,
-  {
-    containerWidth: number;
-    containerHeight: number;
-  }
-> {
+type NodesInColumns = { [key: number]: SNode[] };
+type NodeColors = { fillColor: string; borderColor: string };
+
+export interface ISankeyChartState extends IBasestate {
+  containerWidth: number;
+  containerHeight: number;
+  selectedState: boolean;
+  selectedLinks: Set<Number>;
+  selectedNodes: Set<Number>;
+  selectedNode?: SNode;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  nodeElement?: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  linkElement?: any;
+  selectedLink?: SLink;
+  shouldOverflow: boolean;
+}
+
+const NON_SELECTED_NODE_AND_STREAM_COLOR: string = '#757575';
+const DEFAULT_NODE_COLORS: NodeColors[] = [
+  { fillColor: '#00758F', borderColor: '#002E39' },
+  { fillColor: '#77004D', borderColor: '#43002C' },
+  { fillColor: '#4F6BED', borderColor: '#3B52B4' },
+  { fillColor: '#937600', borderColor: '#6D5700' },
+  { fillColor: '#286EA8', borderColor: '#00457E' },
+  { fillColor: '#A43FB1', borderColor: '#7C158A' },
+  { fillColor: '#CC3595', borderColor: '#7F215D' },
+  { fillColor: '#0E7878', borderColor: '#004E4E' },
+  { fillColor: '#8764B8', borderColor: '#4B3867' },
+  { fillColor: '#9C663F', borderColor: '#6D4123' },
+];
+
+const MIN_HEIGHT_FOR_DOUBLINE_TYPE = 36;
+const MIN_HEIGHT_FOR_TYPE = 24;
+const REST_STREAM_OPACITY: number = 1;
+const NON_SELECTED_OPACITY: number = 1;
+const SELECTED_STREAM_OPACITY: number = 0.3;
+const NON_SELECTED_STREAM_BORDER_OPACITY: number = 0.5;
+const DEFAULT_TEXT_COLOR: string = '#323130';
+const NON_SELECTED_TEXT_COLOR: string = '#FFFFFF';
+
+export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyChartState> {
   private _classNames: IProcessedStyleSet<ISankeyChartStyles>;
   private chartContainer: HTMLDivElement;
   private _reqID: number;
+  private _calloutId: string;
+  private _linkId: string;
```

Comment:
Is this selected link id.

Reply:
This is the link_id that is used in generating the gradient id and id of the path element .